Example/Tests/PubSubUnitTests.swift

```diff
@@ -0,0 +1,521 @@
+        var noWriteKeys: [String]!
+
+        let defaultTimeout: TimeInterval = 10
+
+        let bundle = Bundle(for: type(of: self))
+        if let path = bundle.path(forResource: "Keys", ofType: "plist") {
+
+            if let dict = NSDictionary(contentsOfFile: path) as? [String: Any] {
+                readKey  = dict["readKey"] as? String
+                writeKey = dict["writeKey"] as? String
+                adminKey = dict["adminKey"] as? String
+            }
+
+            allKeys     = [readKey, writeKey, adminKey]
+            noReadKeys  = [writeKey, adminKey]
+            noWriteKeys = [readKey, adminKey]
+        }
+
+        let defaultOptions = PubSubOptions.defaultOptions
+
+        describe("Cogs PubSub Service") {
+
+            describe("get sessionUUID") {
+
+                xit("returns sessionUUID") {
+                    waitUntil(timeout: defaultTimeout) { done in
+                        let socket = MockPubSubSocket(keys: allKeys, options: defaultOptions)
+                        let connectionHandle = PubSubService.connect(socket: socket)
+                        connectionHandle.onNewSession = { _ in
+                            connectionHandle.getSessionUuid() { outcome in
+                                switch outcome {
+                                case .pubSubSuccess(let object):
+                                    if let uuid = object as? String {
+                                        expect(uuid).toNot(beNil())
+                                        expect(uuid).toNot(beEmpty())
+                                    } else {
+                                        fail("Expected String, got \(object)")
+                                    }
+
+                                    connectionHandle.close()
+                                    
+                                    done()
+
+                                case .pubSubResponseError(let errorResponse):
+
+                                    expect(errorResponse.action == PubSubAction.sessionUuid.rawValue).to(beTruthy())
+                                    expect(errorResponse.code).toNot(equal(PubSubResponseCode.success.rawValue))
+
+                                    connectionHandle.close()
+                                    done()
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            describe("channel subcriptions") {
+                let testChannel = "Test"
+
+                describe("subscribe to a channel") {
+
+                    context("when read key is supplied") {
+
+                        it("returns subscribed channels list") {
+                            waitUntil(timeout: defaultTimeout) { done in
+                                let socket = MockPubSubSocket(keys: allKeys, options: defaultOptions)
+                                let connectionHandle = PubSubService.connect(socket: socket)
```

Comment:
In the pull request PubSubConnectionHandle constructor's access level is changed from public to internal bu we can still create instance in the test target. Do you mean we have to revert changes in PubSubService and directly create instances of PubSubConnectionHandle with a MockPubSubSocket for the all tests?  In that case PubSubConnectionHandle will be inaccessible outside of the CogsSDK module but will be testable in the test targets without using PubSubService as a static class which returns PubSubConnectionHandle instance.